### PR TITLE
Make core.internal.convert.toUbyte in CTFE for arrays work with reference type elements and not call postblits/dtors

### PR DESCRIPTION
**Pull request by n8sh**

### CODE REVIEW

**Comment by n8sh** [comment]
Don't merge yet. Adding fix for enums.

**Comment by n8sh** [comment]
Changes done. Please re-review.

File: test/hash/src/test_hash.d [hunk 1]
```diff
@@ -195,6 +196,30 @@ void issue19568()
...
+        {
+            import core.stdc.stdio : puts;
+            if (x) puts("impure");
+        }
+    }
+    enum b2 = () {
+            S[10] a;
+            return ((const S[] a) @nogc nothrow pure @safe => toUbyte(a))(a);
+        }();
+}
```
Comment (ZombineDev) [inline test/hash/src/test_hash.d]: As far the compiler is concerned, `b1` and `b2` are not used, so there's not need to do any semantic analysis on the values they refer to. You should assert the values of `b1` and `b2` are what you expect them to be.

Comment (n8sh) [reply test/hash/src/test_hash.d]: I think the test is right since the test doesn't compile without this PR: https://run.dlang.io/is/v2dXCy

**Pull request merged**